src/GenAPI/Microsoft.DotNet.GenAPI/CSharpFileBuilder.cs

```diff
@@ -91,8 +92,8 @@ public void WriteAssembly(IAssemblySymbol assemblySymbol)
             compilationUnit = compilationUnit.NormalizeWhitespace(eol: Environment.NewLine);
 
             Document document = project.AddDocument(assemblySymbol.Name, compilationUnit);
-            document = Simplifier.ReduceAsync(document).Result;
-            document = Formatter.FormatAsync(document, DefineFormattingOptions()).Result;
+            document = Task.Run(() => Simplifier.ReduceAsync(document)).Result;
+            document = Task.Run(() => Formatter.FormatAsync(document, DefineFormattingOptions())).Result;
 
             document.GetSyntaxRootAsync().Result!
```

Comment:
Looks like you missed this one.   @ViktorHofer I wonder if we should make this method Async instead.

Reply:
And the outer method (in GenAPIApp) that calls this method, as well? And then what about the outer outer outer method?...

Reply:
Probably only a couple layers to this (Main|Execute & GenAPIApp.Run) -- if the entrypoint supports async it could be plumbed all the way, if not we can decide at what layer to do stop.  I was just suggesting that _maybe_ here is not the right layer - we could make this async and use await and then decide at a higher layer (if at all).  My recollection is that the cost of a Task.Run is a lot higher than await -- especially if the work completes synchronously.  I don't think msbuild tasks can have async Execute so that's one point to stop, console apps can have async main though.